### ISSUE TITLE
Docker + TURN not working

### ISSUE THREAD

**Issue by mbleichner** [screego/server#27]
In our company, direct traffic between PCs is blocked, so we have to use the built-in TURN server.

This works fine when I'm using the binary, but not with the docker image (same configuration).

When inspecting port 3478 with tcpdump, it seems like some (not all) UDP packets use an internal docker IP address, which looks a bit suspicious to me.

```
  screego:
    image: screego/server:1.0.1
    ports:
      - mode: host
        target: 5050
        published: 5050
      - mode: host
        protocol: tcp
        target: 3478
        published: 3478
      - mode: host
        protocol: udp
        target: 3478
        published: 3478
    environment:
      - SCREEGO_EXTERNAL_IP=X.X.151.230
      - SCREEGO_SERVER_ADDRESS=0.0.0.0:5050
      - SCREEGO_TURN_ADDRESS=0.0.0.0:3478
      - SCREEGO_AUTH_MODE=none
      - SCREEGO_SERVER_TLS=true
      - SCREEGO_TLS_CERT_FILE=/etc/letsencrypt/live/xxx/fullchain.pem
      - SCREEGO_TLS_KEY_FILE=/etc/letsencrypt/live/xxx/privkey.pem
      - SCREEGO_LOG_LEVEL=debug
    volumes:
      - /etc/letsencrypt:/etc/letsencrypt:ro
```

**Comment by jmattheis** [comment]
Ahh, I think the turn server allocates random ports relay connections. For now network_mode: host should be enough. I've adjusted the docker examples in https://screego.net/#/install?id=docker

**Comment by mbleichner** [comment]
Ah  thanks, makes sense.

The amount of ports needed also seems to be a problem for other TURN servers in docker: https://hub.docker.com/r/instrumentisto/coturn

In this case, they added options `min-port` and `max-port` to mitigate the problem. Would this also be possible here?

I'm not sure if pion/turn supports this, though. There is a GitHub issue saying it's possible, but I'm not sure the solution is actually feasible: https://github.com/pion/turn/issues/129

**Comment by jmattheis** [comment]
Yeah, I've thought of this too. I'll probably give the issue a shot, and implement the address generator for port ranges.

**Comment by tlambertz** [comment]
I just tested the changes in the new release. Gave the turn server 1000 ports and it works great! Thank you :+1: 

(Not using docker, but needed this setting to open the correct ports in my firewall)

**Comment by ThelloD** [comment]
I've spent quite some time testing v1.1.1 in a Docker environment according to the "Port Range" section of the [installation guide](https://screego.net/#/install?id=docker). I think with the current version it's (nearly?) impossible to get Screego to work outside of a local network with port mappings such as:
```
    ports:
      - 5050:5050
      - 3478:3478
      - 50000-50100:50000-50100/udp
```

Locally it work's fine, but behind a NAT both STUN as well as TURN fail and the stream remains black.
The only solution I found to get it working with Docker was to use the `network_mode: host` syntax.
(Other potential solutions approaches might involve additional firewall rules or explicitly specifying *every* in `ports:` with the `- mode: host` syntax since docker-compose does not support ranges here, but I didn't check these.)

I'm not really sure why the issue was closed but at least for me it's not working.

Further, I've tested Screego with STUN in various scenarios, all except one worked well. However, app.screego.net didn't work at all as soon as at least one NAT was in-between the connection path. If the demo instance is hosted using via Docker this might be caused by the same issue.
